### ISSUE TITLE
Feature request: define more specific types for decorators

### ISSUE THREAD

**Issue by dreamorosi** [aws-powertools/powertools-lambda-typescript#302]
## Description of the feature request

As discussed internally, we need to define more specific types for the method decorator that currently uses the `any` type:
```ts
// TODO: Revisit type below & make it more specific
// eslint-disable-next-line  @typescript-eslint/no-explicit-any
type MethodDecorator = (target: any, propertyKey: string | symbol, descriptor: TypedPropertyDescriptor<any>) => any;
``` 

### Related issues, RFCs

[#107](https://github.com/awslabs/aws-lambda-powertools-typescript/pull/107)

**Comment by dreamorosi** [comment]
This issue requires further investigation and depending on the outcome, the current type might end up staying.

The new potential type should be able to accept any `sync` or `async` function. If this is not possible, we should document the reasoning here and remove the `#TODO` from the code.

If anyone is interested in picking this up, please leave a comment here, then discuss your findings and proposed changes before opening a PR.

**Comment by dreamorosi** [comment]
Hi @dVp007 thank you for offering to help on this issue.

Before starting to work on a PR, could you please write a short proposal of the implementation/changes in this issue so that we can discuss it and agree on the changes?

Once we are aligned on the plan, you can open a PR.

**Comment by dreamorosi** [comment]
Closing this in favor of #1204
